src/main/java/com/commercetools/sync/inventories/InventorySync.java

```diff
@@ -298,64 +374,24 @@ public InventorySyncStatistics getStatistics() {
-     */
-    @SuppressFBWarnings("NP_NONNULL_PARAM_VIOLATION") // https://github.com/findbugsproject/findbugs/issues/79
-    private CompletionStage<Void> attemptUpdate(final InventoryEntry entry, final InventoryEntryDraft draft) {
-        final Optional<InventoryEntryDraft> fixedDraft = replaceChannelReference(draft);
-        if (fixedDraft.isPresent()) {
-            final List<UpdateAction<InventoryEntry>> updateActions =
-                InventorySyncUtils.buildActions(entry, fixedDraft.get(), syncOptions, typeService);
-            if (!updateActions.isEmpty()) {
-                return inventoryService.updateInventoryEntry(entry, updateActions)
-                    .thenAccept(updatedEntry -> statistics.incrementUpdated())
-                    .exceptionally(exception -> {
-                        statistics.incrementFailed();
-                        syncOptions.applyErrorCallback(format(CTP_INVENTORY_ENTRY_UPDATE_FAILED, draft.getSku(),
-                            SkuChannelKeyTuple.of(draft).getKey()), exception);
-                        return null;
-                    });
-            }
-        } else {
-            statistics.incrementFailed();
-        }
-        return CompletableFuture.completedFuture(null);
-    }
-
-    /**
-     * Tries to create Inventory Entry in CTP project, using {@code draft}.
-     * Before calling API, the channel reference from {@code draft} is replaced, so it points to proper channel ID
-     * in target system.
-     * It either creates inventory entry and increments "created" counter in statistics, or increments "failed" counter
-     * and executes error callback function in case of any exception.
-     *
-     * @param draft draft of new inventory entry
-     * @return {@link CompletionStage} instance that indicates method progress
+     * @return {@link CompletionStage} of {@link CompletionStage} that may contain passed {@code entry}, updated
+     *      {@link InventoryEntry}, or exception
      */
     @SuppressFBWarnings("NP_NONNULL_PARAM_VIOLATION") // https://github.com/findbugsproject/findbugs/issues/79
-    private CompletionStage<Void> attemptCreate(final InventoryEntryDraft draft) {
-        final Optional<InventoryEntryDraft> fixedDraft = replaceChannelReference(draft);
-        if (fixedDraft.isPresent()) {
-            return inventoryService.createInventoryEntry(fixedDraft.get())
-                .thenAccept(createdEntry -> statistics.incrementCreated())
-                .exceptionally(exception -> {
-                    statistics.incrementFailed();
-                    syncOptions.applyErrorCallback(format(CTP_INVENTORY_ENTRY_CREATE_FAILED, draft.getSku(),
-                        SkuChannelKeyTuple.of(draft).getKey()), exception);
-                    return null;
-                });
-        } else {
-            statistics.incrementFailed();
+    private CompletionStage<InventoryEntry> attemptUpdate(final InventoryEntry entry, final InventoryEntryDraft draft) {
```

Comment:
actually the method with name starting with attempt is kind of wrong
it implies that operation might fail but failure is exceptional situation so it should not
be indicated by method's name. Just 'update' is enough